### ISSUE TITLE
Liquid: filter conferences by current date

### ISSUE THREAD

**Issue by wdconinc** [eicug/eicug.github.io#7]
Right now we have 
```ruby
{%- if include.year=='' -%}
{%- assign selection=site.data.conferences | where: "status", include.status -%}
{%- else -%}
{%- assign selection=site.data.conferences | where: "status", include.status | where: "year", include.year -%}
{%- endif %}
```
which requires `status` to be `upcoming` or `past` and needs changes to `_data` to get updated. A more scalable approach may be to add a field `show_until` and then use a `where_exp` based on current date.

**Comment by wdconinc** [comment]
Ref: https://github.com/eicug/eicug.github.io/blob/main/_includes/conf_table.md

**Comment by buddhasystem** [comment]
I think the current setup gives us a little more flexibility. We can of course define a time window etc. Wouldn't this be an overkill?

**Comment by wdconinc** [comment]
More worried about how it looks when a conference of several months ago is still listed as 'upcoming'. Maybe we can have `status: 'auto'` and only then use date.

**Comment by buddhasystem** [comment]
Good point. I think we should do automatic expiry based on current date. There is some footwork but it's doable.

**Comment by wdconinc** [comment]
Proposal: add `month` and `day` fields in addition to `year`, and use these for 'upcoming', 'past', but also for a `sort` filter.
